akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -256,10 +256,12 @@ object ClusterSingletonManager {
       }
       override def postStop(): Unit = cluster.unsubscribe(self)
 
-      def matchingRole(member: Member): Boolean = role match {
+      private val selfTeam = "team-" + cluster.settings.Team
+
+      def matchingRole(member: Member): Boolean = member.hasRole(selfTeam) && (role match {
         case None    ⇒ true
         case Some(r) ⇒ member.hasRole(r)
-      }
+      })
```

Comment:
Instead of a pattern match I think I'd find `role.forall(member.hasRole(_))` slightly more readable, but I suppose the patternmatch saves a `Function` allocation here?